cdap-explore/src/main/java/co/cask/cdap/hive/serde/ObjectTranslator.java

```diff
@@ -106,7 +106,7 @@ private static Object deserializePrimitive(Object primitive, PrimitiveTypeInfo t
     switch (typeInfo.getPrimitiveCategory()) {
       case STRING:
         // URI, URL, and String all get mapped to string
-        return primitive.toString();
+        return (primitive == null ? null : primitive.toString());
```

Comment:
The other entries do an instance of checks for other types, and we will return null if null is passed in.